Client/Ecosia/IncentivizedSearchChecker/IncentivisedSearchChecker.swift

```diff
@@ -0,0 +1,17 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Core
+
+struct IncentivisedSearchChecker {
```

Comment:
I wonder if this struct should live in core instead. That way Core only needs to expose the Composite, and not the internals. WDYT?

Reply:
My two 🪙 :
I see som value in exposing the composite directly formed by the defined Unleash + Local. Although the purpose of it was to leave freedom to the client on deciding the primary and fallback component to utilise, without having to ask for Core changes. I'm conscious of the fact that this is purely a design perspective, as, concretely, we will have Unleash + Local components only.

Reply:
Thanks for the thoughts 🙏 
Will focus this discussion on https://github.com/ecosia/ios-core/pull/104 and update this PR as needed later.